src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -413,7 +412,21 @@ ExecuteCascadeOperationForRelationIdList(List *relationIdList,
 			{
 				if (!IsCitusTable(relationId))
 				{
-					CreateCitusLocalTable(relationId, cascadeViaForeignKeys);
+					if (PartitionedTable(relationId))
+					{
+						CreatePartitionedCitusLocalTable(relationId,
+														 cascadeViaForeignKeys);
+					}
+					else if (PartitionTable(relationId))
```

Comment:
I'm not sure if we should support this or not. 

Example:
There's a foreign key between Table A and Table B.
Table A is getting converted to a Citus Local Table. Table B is a partition. To cascade the conversion through the foreign key, calling `CreatePartitionedCitusLocalTable` with TableB's parent table doesn't seem correct to me. I think we should error out in such cases. 

However, we should keep the previous if statement above, as we support cascading to a parent table, and then triggering the conversion of all it's child tables.

What do you think @onderkalaci ?